addons/edi/models/edi_transfer.py

```diff
@@ -175,20 +178,29 @@ def do_transfer(self, conn):
         if self.allow_send:
             self.send_outputs(conn)
 
-        _logger.info("%s transfer complete", self.gateway_id.name)
+        self.log_progress(doc, "%s transfer complete" %self.gateway_id.name)
 
     def do_process(self):
         """Process documents"""
         self.ensure_one()
         # Prepare and execute documents, if applicable
         if self.allow_process:
             for doc in self.doc_ids:
-                _logger.info("%s preparing %s", self.gateway_id.name, doc.name)
-                prepared = doc.action_prepare()
-                if prepared:
-                    _logger.info("%s executing %s", self.gateway_id.name, doc.name)
-                    executed = doc.action_execute()
+                if doc.doc_type_id.processing_level in ("prepare", "execute"):
```

Comment:
Can users still manually process if the processing level is disabled?

Reply:
Yes. I believe the processing levels only affect automatic processing by gateway.